src/main/scala/camp/Task3.scala

```diff
@@ -30,6 +31,62 @@ trait IotDeviceRepository[F[_]] {
   def getByUser(userId: Long): F[Seq[IotDevice]]
 }
 
+class UserRepositoryInMemory extends UserRepository[Id]{
+  private var users: Map[String, User] = Map()
+
+  override def registerUser(username: String): Id[User] = {
+    users += (username -> User(users.size.toLong+1,username))
+    users.last._2
```

Comment:
this is not correct, Map implementation changes with it's size:
```scala
var r = Map[String, String]()

r.+=("1" -> "1")
r.getClass.getSimpleName // scala.collection.immutable.Map$Map1

r.+=("2" -> "2")
r.getClass.getSimpleName  // scala.collection.immutable.Map$Map2

r.+=("3" -> "3")
r.getClass.getSimpleName  // scala.collection.immutable.Map$Map3

r.+=("4" -> "4")
r.getClass.getSimpleName  // scala.collection.immutable.Map$Map4

r.+=("5" -> "5")
r.getClass.getSimpleName  // scala.collection.immutable.HashMap$HashTrieMap

r.last._2 // returns 3 instead of 5
```